### PR TITLE
BMO e2e merge before build

### PR DESCRIPTION
**Pull request by lentzi90**
We should merge the PR branch to the base branch before we start the build.

### CODE REVIEW

**Review by kashifest — COMMENTED** [review]
/approve

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

File: jenkins/jobs/bmo_e2e_tests.pipeline [hunk 1]
```diff
@@ -36,6 +36,7 @@ pipeline {
                   extensions: [[$class: "WipeWorkspace"],
                   [$class: "CleanCheckout"],
                   [$class: "CleanBeforeCheckout"],
+                  [$class: "PreBuildMerge", options: [mergeTarget: "main"]],
```
Comment (kashifest) [inline jenkins/jobs/bmo_e2e_tests.pipeline:39]: btw isn't this the same pipeline used for older release branches as well ? In that case, this is not correct, we need to identify the base branch. 
/hold

Comment (lentzi90) [reply jenkins/jobs/bmo_e2e_tests.pipeline:39]: Good point! We actually have e2e tests for release-0.5. I'll update

**Pull request merged**